.github/workflows/soak-testing.yml

```diff
@@ -0,0 +1,191 @@
+      - name: Extract metadata (tags, labels) for Docker
+        id: docker-image-metadata
+        uses: docker/metadata-action@98669ae865ea3cffbcbaa878cf57c20bbf1c6c38
+        with:
+          images: ghcr.io/${{ github.repository }}-${{ env.IMAGE_SUFFIX }}
+          tags: |
+            type=schedule
+            type=ref,event=branch
+            type=ref,event=tag
+            type=ref,event=pr
+      # TODO: Instead of building, publishing, and pulling a Sample App Docker IMage in `docker-compose`. Could we
+      # just build the image in `docker-compose` directly using `build/context`? We would lose the public Docker Image
+      # but we would avoid polluting the published packages and anyone could build from the commit if they wanted to.
+      - name: Build and Push Docker image
+        uses: docker/build-push-action@v2
+        with:
+          push: true
+          context: integration-test-apps/${{ matrix.instrumentation-type}}-instrumentation/${{ matrix.app-platform }}
+          tags: ${{ steps.docker-image-metadata.outputs.tags }}
+          labels: ${{ steps.docker-image-metadata.outputs.labels }}
+          cache-from: type=local,src=/tmp/.buildx-cache
+          cache-to: type=local,dest=/tmp/.buildx-cache
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: ${{ env.AWS_DEFAULT_REGION }}
+      - name: Run All Docker Containers - Sample App + OTel Collector + Load Generator + Alarm Poller
+        id: check-failure-during-performance-tests
+        continue-on-error: true
+        working-directory: .github/docker-performance-tests
+        env:
+          APP_IMAGE: ${{ fromJSON(steps.docker-image-metadata.outputs.json).tags[0] }}
+          INSTANCE_ID: ${{ github.run_id }}-${{ github.run_number }}
+          LISTEN_ADDRESS: 0.0.0.0:8080
+          LOG_GROUP_NAME: otel-sdk-performance-tests
+          LOGS_NAMESPACE: ${{ env.LOGS_NAMESPACE }}
+          LOG_STREAM_NAME: ${{ env.IMAGE_SUFFIX }}
+          PROCESS_COMMAND_LINE_DIMENSION_VALUE: ${{ env.PROCESS_COMMAND_LINE_DIMENSION_VALUE }}
+          PROCESS_EXECUTABLE_NAME: ${{ env.PROCESS_EXECUTABLE_NAME }}
+          HOSTMETRICS_INTERVAL_SECS: ${{ env.HOSTMETRICS_INTERVAL_SECS }}
+          TEST_DURATION_MINUTES: ${{ env.TEST_DURATION_MINUTES }}
+          NUM_OF_CPUS: ${{ env.NUM_OF_CPUS }}
+          CPU_LOAD_THRESHOLD: ${{ env.CPU_LOAD_THRESHOLD }}
+          TOTAL_MEMORY_THRESHOLD: ${{ env.TOTAL_MEMORY_THRESHOLD }}
+          AWS_DEFAULT_REGION: ${{ env.AWS_DEFAULT_REGION }}
+        run: >-
+          docker-compose up;
+          exit $(docker inspect $(docker ps --quiet --all --filter "name=docker-performance-tests_alarms-poller") --format="{{.State.ExitCode}}");
+      - name: Install script dependencies
+        run: pip install boto3
```

Comment:
This is one of the downsides of using Python scripts. It lends itself to be less vendor agnostic I would say? I couldn't figure out how to download `boto3` as part of the scripts that needed it hence I left it as a step of the workflow here.